app/views/users/show.html.erb

```diff
@@ -24,15 +24,106 @@
     }
     <% end %>
   </style>
-  <div class="user-profile-header" style="<%= user_colors_style(@user) %>" itemscope itemtype="http://schema.org/Person" itemprop="mainEntity">
-    <meta itemprop="url" content="https://dev.to/<%= @user.username %>">
+
+<script type="application/ld+json">
+  {
```

Comment:
I wonder if this wouldn't be better built in Ruby in the controller and thenput here as a variable, like

```html
<script type="application/ld-json">
<%= @user_json_ld %>
<script>
```

if you see my comments you'll notice how complicated is to insert if conditon while you're basically building a JSON string. 

I'd build a hash in Ruby and then call `.to_json` in it. 

What do you think?

Reply:
I understand what your concern is and I think that that's not a bad idea!